packages/ui-components/src/components/form/RadioGroup.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <div>
+    <div class="sr-only">{{ label }}</div>
+    <div class="flex flex-col gap-2">
+      <div
+        v-for="(option, index) in options"
+        :key="index"
+        class="flex items-center gap-3"
+      >
+        <input
+          :id="`option-${index}`"
+          v-model="selected"
+          type="radio"
+          :value="option"
+          class="cursor-pointer h-3 w-3 bg-foundation border-foundation ring-[1.5px] ring-primary checked:ring-primary-focus focus:ring-offset-foundation focus:outline-primary-focus"
+        />
+        <label
+          class="flex items-center gap-1.5 text-sm text-foreground cursor-pointer"
+          :for="`option-${index}`"
+        >
+          <Component :is="option.icon" v-if="option.icon" class="h-9 w-9" />
+          {{ option.title }}
+        </label>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, watchEffect } from 'vue'
+import type { ConcreteComponent } from 'vue'
+
+interface Option {
+  title: string
+  icon?: ConcreteComponent
+}
+
+const props = defineProps<{
+  label: string
+  options: Option[]
+}>()
+
+const emit = defineEmits(['update:selected'])
```

Comment:
Use `defineModel` to enable devs to pass in an existing `selected` state and also be able to emit out an update event when writing to the model ref.

Reply:
Also - instead of the model holding `Option` it should just hold the ID of the option (a string), otherwise, if you want to preselect an option, you need to have its entire `Option` object which contains irrelevant data for selection like the icon it has.

This means that you should also introduce an `id` prop in the `Option` interface

Reply:
Also - there doesn't seem to be any `vee-validate` support here, no way to set the `name` of the form field, feed in rules or ensure `vee-validate` can even read/write the selected state.

(Not sure, but you might not be able to use `defineModel` and might have to define the prop and event manually)

Read more on how to implement vee-validate support: https://vee-validate.logaretm.com/v4/examples/custom-inputs/
And feel free to look at other components like FormSelectBase to see useful props to add and how to integrate into vee-validate

Reply:
I've removed the new RadioGroup component. I created a new component called Radio which is a lot closer to how the other items work. It is based on checkbox so should be a lot more consistent.